client/pruning/main.go

```diff
@@ -0,0 +1,91 @@
+// specified by command flags.
+func PruningCmd(appCreator servertypes.AppCreator) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "prune",
+		Short: "Prune app history states based on the pruning options specified by flags",
+		Long: `Prune app history states based on the pruning options specified by flags,
+		Pruning options can be provided via the '--pruning' flag or alternatively with '--pruning-keep-recent', and
+		'--pruning-interval' together.
+		
+		For '--pruning' the options are as follows:
+		
+		default: the last 362880 states are kept, pruning at 10 block intervals
+		nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
+		everything: 2 latest states will be kept; pruning at 10 block intervals.
+		custom: allow pruning options to be manually specified through 'pruning-keep-recent', and 'pruning-interval'.
+
+		besides pruning options, database home directory and database backend type should also be specified via flags
+		'--home' and '--app-db-backend'.
+		valid app-db-backend type includes 'goleveldb', 'cleveldb', 'rocksdb', 'boltdb', and 'badgerdb'.
+		`,
+		Example: "prune --home './' --app-db-backend 'goleveldb' --pruning 'custom' --pruning-keep-recent 100 --pruning-interval 10",
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			vp := viper.New()
+
+			// Bind flags to the Context's Viper so the app construction can set
+			// options accordingly.
+			if err := vp.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+
+			home := vp.GetString(flags.FlagHome)
+			db, err := openDB(home, server.GetAppDBBackend(vp))
+			if err != nil {
+				return err
+			}
+
+			logger, _ := log.NewDefaultLogger("plain", "info", false)
+			// we should set pruning options in providerCreator
+			app := appCreator(logger, db, nil, vp)
+			cms := app.CommitMultiStore()
+			cmsOptions := cms.GetPruning()
+			// set pruning options for cms in case we forgot to apply the pruning options in providerCreator
+			if cmsOptions.Strategy == 0 && cmsOptions.KeepRecent == 0 {
+				pruningOptions, err := server.GetPruningOptionsFromFlags(vp)
+				if err != nil {
+					return err
+				}
+				cms.SetPruning(pruningOptions)
+			}
+
+			if rootMultiStore, ok := cms.(*rootmulti.Store); ok {
+				err = rootMultiStore.PruneHistoryVersions()
+				return err
+			}
+
+			return fmt.Errorf("currently only support the pruning of rootmulti.Store type")
+		},
+	}
+
+	cmd.Flags().String(flags.FlagHome, "", "The database home directory")
+	cmd.Flags().String(server.FlagPruning, pruningtypes.PruningOptionDefault, "Pruning strategy (default|nothing|everything|custom)")
+	cmd.Flags().Uint64(server.FlagPruningKeepRecent, 0, "Number of recent heights to keep on disk (ignored if pruning is not 'custom')")
+	cmd.Flags().Uint64(server.FlagPruningInterval, 0, "Height interval at which pruned heights are removed from disk (ignored if pruning is not 'custom')")
+	cmd.Flags().String(flags.FlagDBType, "", "the backend db type")
```

Comment:
```suggestion
	cmd.Flags().String(flags.FlagDBType, "", "The type of database for application and snapshots databases")
```

the `StartCmd` should have this flag too, but that can be done in other PRs ;D